File: src/nuitrack_bridge/types/hand_data.rs
#[cxx::bridge(namespace = "nuitrack_bridge::hand_data")]
pub mod ffi {

    #[namespace = "tdv::nuitrack"]
    unsafe extern "C++" {
        type HandData;
        type UserHands = crate::nuitrack_bridge::types::hand::ffi::UserHands;
    }

    impl SharedPtr<HandData> {}

    unsafe extern "C++" {
        include!("nuitrack_bridge/types/hand_data.h");

        #[cxx_name = "getTimestamp"] // Corrected
        pub fn timestamp(data: &HandData) -> Result<u64>;
        #[cxx_name = "getNumUsers"] // Corrected
        pub fn num_users(data: &HandData) -> Result<i32>;
        
        #[cxx_name = "getUsersHands"] // Corrected
        pub fn users_hands(data: &HandData) -> Result<UniquePtr<CxxVector<UserHands>>>;
    }
}

unsafe impl Send for ffi::HandData {}
unsafe impl Sync for ffi::HandData {}

File: src/nuitrack_bridge/types/hand_data.cc
#include "nuitrack_bridge/types/hand_data.h"
#include "nuitrack/types/HandTrackerData.h"
#include "nuitrack/types/Hand.h"           

namespace nuitrack_bridge::hand_data {

    uint64_t getTimestamp(const HandData& data) {
        // Assuming Nuitrack methods can throw, wrap in try-catch if necessary,
        // similar to skeleton_tracker.cc for robustness. For brevity, not shown here
        // but should be considered for production code.
        return data.getTimestamp();
    }

    int32_t getNumUsers(const HandData& data) {
        return static_cast<int32_t>(data.getNumUsers());
    }

    std::unique_ptr<std::vector<UserHands>> getUsersHands(const HandData& data) {
        return std::make_unique<std::vector<UserHands>>(data.getUsersHands());
    }

}